aptos-move/framework/aptos-framework/sources/block.move

```diff
@@ -178,6 +180,31 @@ module aptos_framework::block {
         );
     }
 
+    ///  Emit a `NewBlockEvent` event. This function will be invoked by write set script directly to generate the
+    ///  new block event for WriteSetPayload.
+    public fun emit_writeset_block_event(framework_signer: &signer) acquires BlockResource {
+        system_addresses::assert_aptos_framework(framework_signer);
+
+        let block_metadata_ref = borrow_global_mut<BlockResource>(@aptos_framework);
+        block_metadata_ref.height = event::counter(&block_metadata_ref.new_block_events);
+
+        let genesis_id = @0x0;
+        emit_new_block_event(
```

Comment:
Why not reuse emit_genesis_block_event()? That function also doesn't need to take the vm signer as an arg since it doesn't use or check it anyway.

Reply:
Actually hold off on this a bit as the values in the new block event might not be valid.

Reply:
`emit_genesis_block_event` takes a `signer` instead of a `&signer` so not sure what you mean by that.